### PR TITLE
docs - awscc_amplify_branch

### PR DESCRIPTION
**Pull request by novekm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

Relates OR Closes #0000

### CODE REVIEW

File: templates/resources/amplify_branch.tmpl [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
```
Comment (jackywong-amazon) [inline templates/resources/amplify_branch.tmpl:1]: the file name should be called amplify_branch.md.tmpl

File: docs/resources/amplify_branch [hunk 2]
```diff
@@ -0,0 +1,169 @@
+---
```
Comment (jackywong-amazon) [inline docs/resources/amplify_branch:1]: This doco is automatically generated using the doco tool.

File: examples/resources/awscc_amplify_branch/multi_branch.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+      name  = "Environment"
+      value = "PROD"
+    },
+  ]
+}
+
+resource "awscc_amplify_branch" "dev" {
+  app_id      = awscc_amplify_app.example.app_id
+  branch_name = "main"
```
Comment (jackywong-amazon) [inline examples/resources/awscc_amplify_branch/multi_branch.tf:26]: Would it be better to use development branch name as an example?

File: templates/resources/amplify_branch.tmpl (hunk 1 above)
Comment (novekm) [reply templates/resources/amplify_branch.tmpl]: whoops - fixed typo in file name and ran `make docs` again. Should be corrected now

File: docs/resources/amplify_branch (hunk 2 above)
Comment (jackywong-amazon) [reply docs/resources/amplify_branch]: Update the amplift_branch.md.tmpl file, then rerun the make doco command

File: examples/resources/awscc_amplify_branch/multi_branch.tf (hunk 3 above)
Comment (novekm) [reply examples/resources/awscc_amplify_branch/multi_branch.tf:26]: yea, copy/paste error sorry - will fix now

Comment (novekm) [reply examples/resources/awscc_amplify_branch/multi_branch.tf]: Fixed branch name typo and re-ran `make docs` command

**Review by jackywong-amazon — APPROVED** [review]
LGTM, thanks for your contribution !

**Pull request merged**